sphinx/texinputs/footnotehyper-sphinx.sty

```diff
@@ -325,8 +328,39 @@
 % some extras for Sphinx :
 % \sphinxfootnotemark: usable in section titles and silently removed from TOCs.
 \def\sphinxfootnotemark [#1]%
-   {\ifx\thepage\relax\else\protect\spx@opt@BeforeFootnote
-                             \protect\footnotemark[#1]\fi}%
+   {\ifx\thepage\relax\else\sphinxfootref{#1}\fi}%
+% \sphinxfootref: inspired by \footref from footmisc package as
+% re-defined by hyperref, but with an extra FNH@! branch:
+% - \spx@opt@BeforeFootnote is from BeforeFootnote sphinxsetup option
+% - \ref{fn:#1} branch:
+%     the latex.py writer inserts at start of footnotetext contents
+%     \phantomsection\label{fn:<num>}, so we can use \ref which
+%     hyperref converts into an hyperlink. This is basically the
+%     idea of the \footref macro from package footmisc.
+% - FNH@! branch:
+%     explicitly numbered footnotes in latex.py output always use first
+%     the footnote environment, then \sphinxfootnotemark. We patched
+%     \FNH@footnoteenv@i to define the \FNH@!<number> macro as storage
+%     for the internal hyperref target id for the hyperref anchor. So
+%     we can then create an hyperlink from any \sphinxfootnotemark with
+%     this <number> as argument.
```

Comment:
Thus there are different things:
- when footnotetext mark-up is used, hyperref created no target anchor, and we use `\phantomsection` mark-up to add one with a label `\label{fn:number}`. I think this will not coliide with other labels,
- when footnote is used we don't change anything to latex.py output but add a LaTeX hack inside internal macro of footnotehyper-sphinx; this allows then for manually numbered footnotes which are referred to multiple times to be all hyperlinks. This pre-supposes that the first one in latex file is rendered by footnote environment, and all later ones are using `\sphinxfootnotemark` mark-up.